From 55bf4242e5ca77a5912c7956e779ae2d8ec4c080 Mon Sep 17 00:00:00 2001
From: Cheick Traore <cheick.traore@foss.st.com>
Date: Tue, 11 Feb 2025 11:13:52 +0100
Subject: [PATCH] arm64: dts: st: add 3v3 regulator for pwm-backlight on
 stm32mp257f-dk

Add scmi_v3v3 as power-supply for pwm backlight in order to completely
disable and suspend the timer when suspending the display.

Change-Id: Ic47a7b8b5d2be19c05e389ab36d615bc24346f49
Signed-off-by: Cheick Traore <cheick.traore@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/436175
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yannick FERTRE <yannick.fertre@foss.st.com>
Reviewed-by: Yannick FERTRE <yannick.fertre@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
index d1590fecbbeb..e7f9e59b2020 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -196,6 +196,7 @@ panel_lvds_backlight: panel-lvds-backlight {
 		pwms = <&pwm3 1 1000000 0>;
 		brightness-levels = <0 16 22 30 40 55 75 102 138 188 255>;
 		default-brightness-level = <10>;
+		power-supply = <&scmi_v3v3>;
 		status = "okay";
 	};
 
-- 
2.50.1

